src/langchain_google_firestore/vectorstores.py

```diff
@@ -0,0 +1,414 @@
+        elif USER_AGENT not in client_agent:
+            self.client._client_info.user_agent = " ".join([client_agent, USER_AGENT])
+
+        if isinstance(collection, str):
+            self.collection = self.client.collection(collection)
+        else:
+            self.collection = collection
+
+        self.embedding = embedding
+        self.content_field = content_field
+        self.metadata_field = metadata_field
+        self.embedding_field = embedding_field
+        self.distance_strategy = distance_strategy
+        self.filters = filters
+
+    @property
+    def embeddings(self) -> Optional[Embeddings]:
+        return self.embedding
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Add or update texts in the vector store. If the `ids` are provided, and
+        a Firestore document with the same id exists, it will be updated.
+        Otherwise, a new Firestore document will be created.
+
+        Args:
+            texts: The texts to add to the vector store.
+            metadatas: The metadata to add to the vector store. Defaults to None.
+            ids: The document ids to use for the new documents. If not provided, new
+            document ids will be generated.
+
+        Returns:
+            List[str]: The list of document ids added to the vector store.
+        """
+        texts_len = sum(1 for _ in iter(texts))
+        ids_len_match = not ids or len(ids) == texts_len
+        metadatas_len_match = not metadatas or len(metadatas) == texts_len
+
+        assert texts_len != 0, "No texts provided to add to the vector store."
+        assert (
+            metadatas_len_match
+        ), "The length of metadatas must be the same as the length of texts or zero."
+        assert (
+            ids_len_match
+        ), "The length of ids must be the same as the length of texts or zero."
+
+        _ids: List[str] = []
+        db_batch = self.client.batch()
+
+        def get_doc_id(i: int) -> Optional[str]:
+            if ids:
+                return ids[i]
+            return None
+
+        for batch in more_itertools.chunked(texts, WRITE_BATCH_SIZE):
+            texts_embs = self.embedding.embed_documents(batch)
+            for i, text in enumerate(batch):
+                doc_id = get_doc_id(i)
+                doc = self.collection.document(doc_id)
+                _ids.append(doc.id)
+
+                data = {
+                    self.content_field: text,
+                    self.embedding_field: Vector(texts_embs[i]),
+                }
+
+                if metadatas:
+                    data[self.metadata_field] = metadatas[i]
+
+                db_batch.set(doc, data, merge=True)
+
+            db_batch.commit()
+
+        return _ids
+
+    def delete(self, ids: Optional[List[str]] = None, **kwargs: Any) -> None:
```

Comment:
Do we have tests for this? -- I couldn't find it in the unit test

The DocumentReference delete is not the same as this one.

Reply:
Added